### ISSUE TITLE
'ip' is required even if 'host' present in 2.3.3

### ISSUE THREAD

**Issue by Alan-R** [ktbyers/netmiko#1255]
The documentation implies that ip is optional if host is provided (and vice versa).

It appears that at least in 2.3.3 this isn't the case. If you don't supply 'ip', then you get an exception 'XXX' object has no attribute 'ip'.

This code is the culprit:
```
        if ip:
            self.host = ip
            self.ip = ip
        elif host:
            self.host = host
        self.host = self.host.strip()
        self.ip = self.ip.strip()
```

The last line should probably say `self.ip = ip.strip()`, and then the line that says `self.ip = ip` becomes redundant and should be removed.

**Comment by carlniger** [comment]
@Alan-R you must have caught it just at the wrong time! There was about a one day period where this was the case but it got fixed in #1244 -- also sorry that was my bad :)

**Comment by Alan-R** [comment]
Thanks!
